dataset_builders/pie/cdcp/cdcp.py

```diff
@@ -0,0 +1,142 @@
+import dataclasses
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+import datasets
+from pytorch_ie.annotations import BinaryRelation, LabeledSpan
+from pytorch_ie.core import Annotation, AnnotationList, annotation_field
+from pytorch_ie.documents import (
+    TextBasedDocument,
+    TextDocumentWithLabeledSpansAndBinaryRelations,
+)
+
+from pie_datasets import GeneratorBasedBuilder
+from pie_datasets.document.processing.text_span_trimmer import trim_text_spans
+
+log = logging.getLogger(__name__)
+
+
+def dl2ld(dict_of_lists):
+    return [dict(zip(dict_of_lists, t)) for t in zip(*dict_of_lists.values())]
+
+
+def ld2dl(list_of_dicts, keys: Optional[List[str]] = None, as_list: bool = False):
+    return {k: [d[k] for d in list_of_dicts] for k in keys}
+
+
+@dataclasses.dataclass(frozen=True)
+class Attribute(Annotation):
+    value: str
+    annotation: Annotation
+
+
+@dataclasses.dataclass
+class CDCPDocument(TextBasedDocument):
+    propositions: AnnotationList[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationList[BinaryRelation] = annotation_field(target="propositions")
+    urls: AnnotationList[Attribute] = annotation_field(target="propositions")
+
+
+def example_to_document(
+    example: Dict[str, Any],
+    relation_label: Callable[[int], str],
+    proposition_label: Callable[[int], str],

```

Comment:
this is the wrong type annotation (we now pass the label object, not the conversion method anymore)

Reply:
updated in 24be1d1